### PR TITLE
Remove amp-fit-text block setting

### LINKED ISSUE #4557: Deprecate/remove AMP fit-text setting
## Feature description

The `amp-fit-text` setting does not seem relevant after the years since it has been available. I don't recall it being used once by a user. It seems much more suitable for programmatically adding text to a post. For example, it would make great sense to use for auto-generated web stories. But in this context where users are manually entering text, why wouldn't they just set the font size to the desired height on their own? The amp-fit-text ability just seems overly complicated and unnecessary. It should be removed. It also does not appear to even work properly per #5658.

~A paragraph block currently has an “AMP Settings” section among its block settings. It should be eliminated, with the “automatically fit text to container” being moved to the Text settings panel. Users should not need to know about “AMP” when they author content.~

~Also, the `amp-fit-text` should be assured to work when the content is viewed on a non-AMP page.~

image

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation brief

* 

## QA testing instructions

* 

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by pierlon**
## Summary

Removes the `amp-fit-text` setting and related controls from the block panel and also removes the related block attributes from text blocks that had the setting enabled.

Fixes #4557

## Checklist

- [x] My pull request is addressing an open issue (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [x] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: assets/src/block-editor/helpers/index.js [hunk 1]
```diff
@@ -163,68 +130,79 @@ export const addAMPAttributes = ( settings, name ) => {
...
-	const match = contentRegex.exec( content );
-
-	let newContent = content;
-
-	if ( match && match[ 1 ] ) {
-		newContent = match[ 1 ];
+		settings.deprecated.unshift( {
+			supports: settings.supports,
+			attributes: {
+				...settings.attributes,
```
Comment (johnwatkins0) [inline assets/src/block-editor/helpers/index.js]:
```suggestion
				...( settings?.attributes || {} ),
```

**Review by johnwatkins0 — CHANGES_REQUESTED** [review]
One minor suggestion.

File: assets/src/block-editor/helpers/index.js [hunk 2]
```diff
@@ -163,68 +130,79 @@ export const addAMPAttributes = ( settings, name ) => {
...
-	const match = contentRegex.exec( content );
-
-	let newContent = content;
-
-	if ( match && match[ 1 ] ) {
-		newContent = match[ 1 ];
+		settings.deprecated.unshift( {
+			supports: settings.supports,
+			attributes: {
+				...( settings?.attributes || {} ),
```
Comment (johnwatkins0) [inline assets/src/block-editor/helpers/index.js]: On second thought, the `?` wouldn't be needed. 

```suggestion
				...( settings.attributes || {} ),
```

Comment (pierlon) [reply assets/src/block-editor/helpers/index.js]: The outcome would be the same, right? But yea, the `?` is redundant.

**Review by johnwatkins0 — APPROVED** [review]

**Comment by westonruter** [comment]
Testing with this `post_content`:

```html
<!-- wp:paragraph {"ampFitText":true} -->
<amp-fit-text layout="fixed-height" min-font-size="4" max-font-size="172" height="172"><p>paragraph</p></amp-fit-text>
<!-- /wp:paragraph -->

<!-- wp:heading {"ampFitText":true} -->
<amp-fit-text layout="fixed-height" min-font-size="4" max-font-size="172" height="172"><h2>teasdasd</h2></amp-fit-text>
<!-- /wp:heading -->

<!-- wp:code {"ampFitText":true} -->
<amp-fit-text layout="fixed-height" min-font-size="4" max-font-size="172" height="172"><pre class="wp-block-code"><code>test</code></pre></amp-fit-text>
<!-- /wp:code -->

<!-- wp:quote {"ampFitText":true} -->
<amp-fit-text layout="fixed-height" min-font-size="5" max-font-size="73" height="90"><blockquote class="wp-block-quote"><p>quote</p></blockquote></amp-fit-text>
<!-- /wp:quote -->
```

Before the changes in this PR, it looks like this:

Non-AMP | AMP
----------|------
Screen Shot 2020-12-28 at 22 47 31 | image

Upon upgrading to the build in this PR, accessing the editor results in block validation errors for the Code block and the Quote block:

image

image

Interesting the Header block and the Paragraph block automatically strip out the `<amp-fit-text>` tags, but they remain for Quote and Code:

```html
<!-- wp:paragraph -->
<p>paragraph</p>
<!-- /wp:paragraph -->

<!-- wp:heading -->
<h2>teasdasd</h2>
<!-- /wp:heading -->

<!-- wp:code -->
<amp-fit-text layout="fixed-height" min-font-size="4" max-font-size="172" height="172"><pre class="wp-block-code"><code>test</code></pre></amp-fit-text>
<!-- /wp:code -->

<!-- wp:quote -->
<amp-fit-text layout="fixed-height" min-font-size="5" max-font-size="73" height="90"><blockquote class="wp-block-quote"><p>quote</p></blockquote></amp-fit-text>
<!-- /wp:quote -->
```

Is the presence of block validation errors expected? I suppose so. Attempting block recovery for those two blocks is successful:

```html
<!-- wp:paragraph -->
<p>paragraph</p>
<!-- /wp:paragraph -->

<!-- wp:heading -->
<h2>teasdasd</h2>
<!-- /wp:heading -->

<!-- wp:code -->
<pre class="wp-block-code"><code>test</code></pre>
<!-- /wp:code -->

<!-- wp:quote -->
<blockquote class="wp-block-quote"><p>quote</p></blockquote>
<!-- /wp:quote -->
```

And then the result is identical AMP and non-AMP, and no trace of `amp-fit-text`.

Is this all working as intended?

**Comment by westonruter** [comment]
Granted, there are probably no instances (or very few instances) of these blocks being used with fit-text.

**Comment by pierlon** [comment]
> Is the presence of block validation errors expected?

No, it should be be able to deprecate gracefully without any errors. I'll take a look at the example you provided.

**Review by westonruter — CHANGES_REQUESTED** [review]
Per https://github.com/ampproject/amp-wp/pull/5729#issuecomment-751971884

**Comment by pierlon** [comment]
I've fixed the issue in https://github.com/ampproject/amp-wp/pull/5729/commits/bdb65aab8eb3fdf263d4590de84023a83d330f4b. The `code` and `quote` blocks were not being deprecated as expected due to an additional prop being added to the element after the block is saved. I've tracked it down to the [`core/generated-class-name/save-props`](https://github.com/WordPress/gutenberg/blob/f63053cace3c02e284f00918e1854284c85b9132/packages/block-editor/src/hooks/generated-class-name.js#L22) filter which adds a generated `className` prop if the block supports the `className` feature (which all blocks do by default). The `paragraph` and `heading` blocks explicitly do not support the feature, which is why the issue did not occur on those blocks.

**Comment by westonruter** [comment]
Fixed indeed!

Input:

```html
<!-- wp:paragraph {"ampFitText":true} -->
<amp-fit-text layout="fixed-height" min-font-size="4" max-font-size="172" height="172"><p>paragraph</p></amp-fit-text>
<!-- /wp:paragraph -->

<!-- wp:heading {"ampFitText":true} -->
<amp-fit-text layout="fixed-height" min-font-size="4" max-font-size="172" height="172"><h2>teasdasd</h2></amp-fit-text>
<!-- /wp:heading -->

<!-- wp:code {"ampFitText":true} -->
<amp-fit-text layout="fixed-height" min-font-size="4" max-font-size="172" height="172"><pre class="wp-block-code"><code>test</code></pre></amp-fit-text>
<!-- /wp:code -->

<!-- wp:quote {"ampFitText":true} -->
<amp-fit-text layout="fixed-height" min-font-size="5" max-font-size="73" height="90"><blockquote class="wp-block-quote"><p>quote</p></blockquote></amp-fit-text>
<!-- /wp:quote -->
```

Output:

```html
<!-- wp:paragraph -->
<p>paragraph</p>
<!-- /wp:paragraph -->

<!-- wp:heading -->
<h2>teasdasd</h2>
<!-- /wp:heading -->

<!-- wp:code -->
<pre class="wp-block-code"><code>test</code></pre>
<!-- /wp:code -->

<!-- wp:quote -->
<blockquote class="wp-block-quote"><p>quote</p></blockquote>
<!-- /wp:quote -->
```

No block validation errors:

image

**Comment by westonruter** [comment]
@pierlon Question: Is there a reason why you went with `removeClassFromAmpFitTextBlocks` in JS as opposed to using the existing [`ObsoleteBlockAttributeRemover`](https://github.com/ampproject/amp-wp/blob/develop/src/ObsoleteBlockAttributeRemover.php)?

**Comment by pierlon** [comment]
The `ObsoleteBlockAttributeRemover` service would not be applicable in this context as the block deprecation is handled on the client side and so the added `class` attribute will never be seen by the service until the post is saved, and by that time the "This block contains invalid content" notice would already be displayed.

**Comment by westonruter** [comment]
See also successful block migration messages:

> image

**Review by westonruter — APPROVED** [review]

**Pull request merged**